openmetadata-service/src/main/java/org/openmetadata/service/resources/apps/AppResource.java

```diff
@@ -686,6 +687,9 @@ public Response delete(
           @PathParam("name")
           String name) {
     App app = repository.getByName(null, name, repository.getFields("bot,pipelines"));
+    if (app.getSystemApp()) {
+      throw new AuthorizationException("System App cannot be uninstalled");
```

Comment:
can we not use AuthorizationException here or introduce a new exception.
AuthorizationException should be used specifically for permissions not for anything else.

Reply:
changed to BadRequest